### ISSUE TITLE
Work-around for Bugsnag with Angular SSR?

### ISSUE THREAD

**Issue by jack-guy** [bugsnag/bugsnag-angular#14]
### Expected behavior
I'm running an AOT-compiled server-side rendered Angular app with Bugsnag. I understand from [this issue](https://github.com/bugsnag/bugsnag-js/issues/326) that SSR is not supported, but even if I provide the default Angular `ErrorHandler` in a server context, I still get the following error:

### Observed behavior
```
> ts-node server.ts

/node_modules/bugsnag-angular/dist/index.js:11
import { ErrorHandler, Injectable } from "@angular/core";
       ^

SyntaxError: Unexpected token {
    at new Script (vm.js:74:7)
    at createScript (vm.js:246:10)
    at Object.runInThisContext (vm.js:298:10)
    at Module._compile (internal/modules/cjs/loader.js:646:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
```
It appears the distributed Bugsnag code relies on the import being compiled out by the angular compiler or something? Whatever the case, `ts-node` immediately crashes because the import / export isn't supported by Node. The SSR bundles are supposed to be imported using CommonJS requires which means I can't compile out the imports myself either.
```
// Leave this as require() since this file is built dynamically
const { AppServerModuleNgFactory, LAZY_MODULE_MAP } = require('./dist-server/main');
```

### Steps to reproduce
1. Follow Bugsnag Angular integration steps
2. Run a standard SSR app (e.g. [this one](https://github.com/angular/universal-starter))
3. Server application crashes

### Version
* `@angular/core 6.x`, `@angular/express-engine 6.x`

### Additional information
I considered conditionally importing bugsnag based on the `isPlatformBrowser` helper using `import()`, but Angular dependency injection doesn't support factories that return promises.

**Comment by bengourley** [comment]
Hey @jack-guy. Thanks for reporting this issue – this is probably pretty frustrating  for you!

I cloned the [basic SSR app you linked to](https://github.com/angular/universal-starter) and managed to get the app to compile and run successfully with the following changes to app.module.ts:

```js
import BugsnagErrorHandler from 'bugsnag-angular'
import bugsnag from 'bugsnag-js'

import {BrowserModule} from '@angular/platform-browser';
import {ErrorHandler, NgModule, PLATFORM_ID} from '@angular/core';
import {RouterModule} from '@angular/router';
import {isPlatformBrowser} from '@angular/common';

const bugsnagClient = isPlatformBrowser(PLATFORM_ID) ? bugsnag('API_KEY') : null

import { AppComponent } from './app.component';
import { HomeComponent } from './home/home.component';
import {TransferHttpCacheModule} from '@nguniversal/common';

export function errorHandlerFactory() {
  return new BugsnagErrorHandler(bugsnagClient)
}

@NgModule({
  declarations: [
    AppComponent,
    HomeComponent,
  ],
  imports: [
    BrowserModule.withServerTransition({appId: 'my-app'}),
    RouterModule.forRoot([
      { path: '', component: HomeComponent, pathMatch: 'full'},
      { path: 'lazy', loadChildren: './lazy/lazy.module#LazyModule'},
      { path: 'lazy/nested', loadChildren: './lazy/lazy.module#LazyModule'}
    ]),
    TransferHttpCacheModule,
  ],
  providers: isPlatformBrowser(PLATFORM_ID)
    ? [ { provide: ErrorHandler, useFactory: errorHandlerFactory } ]
    : [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

The only downside of this is that bugsnag-js logs a warning when it gets imported on the server, but as long as you don't do anything with it, it's fine.

As far as compiling out the import syntax, it looks like this example app handles it fine and so I think that's something your app should do too. `ts-node` by default might leave out everything it finds in `node_modules`, but I'm not sure.

Let me know if this gets you any further! Rest assured that we are working on a universal js notifier at the moment to support SSR and make the user experience across the different js platforms much more consistent.

**Comment by jack-guy** [comment]
Thanks for the detailed look @bengourley - that use of `isPlatformBrowser` looks similar to what I've got now. I'm guessing since the sample SSR app relies on Webpack for the server code's compilation the imports are resolved just fine. I'd like to keep using `ts-node` instead of webpack in the short term, just for ease of development / source maps, but I ended up just copying the `bugsnag-angular` code into my repo to get it working without issue.
